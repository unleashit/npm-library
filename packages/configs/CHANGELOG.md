# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/unleashit/npm-library/compare/@unleashit/dev@0.1.1...@unleashit/dev@0.1.2) (2023-03-10)

**Note:** Version bump only for package @unleashit/dev





## 0.1.1 (2023-02-24)

**Note:** Version bump only for package @unleashit/dev
