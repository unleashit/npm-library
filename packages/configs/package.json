{
  "name": "@unleashit/configs",
  "version": "0.1.2",
  "files": [
    "index.ts",
    "tsup.config.js"
  ],
  "description": "Development dependencies for npm-library",
  "private": true,
  "scripts": {},
  "license": "MIT",
  "keywords": [
    "login",
    "react login",
    "responsive",
    "custom fields",
    "react",
    "react components",
    "npm packages",
    "npm components",
    "ui components",
    "ui library",
    "monorepo",
    "lerna",
    "typescript"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unleashit/npm-library.git"
  },
  "homepage": "https://github.com/unleashit/npm-library/tree/master/packages/dev",
  "bugs": {
    "url": "https://github.com/unleashit/npm-library/issues"
  },
  "devDependencies": {
    "tsup": "^6.7.0",
    "react": "^18.2.0"
  }
}
