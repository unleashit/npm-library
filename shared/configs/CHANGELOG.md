# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.5](https://github.com/unleashit/npm-library/compare/@unleashit/configs@1.0.4...@unleashit/configs@1.0.5) (2023-12-29)

**Note:** Version bump only for package @unleashit/configs

## [1.0.4](https://github.com/unleashit/npm-library/compare/@unleashit/configs@1.0.3...@unleashit/configs@1.0.4) (2023-12-29)

**Note:** Version bump only for package @unleashit/configs

## [1.0.3](https://github.com/unleashit/npm-library/compare/@unleashit/configs@1.0.1...@unleashit/configs@1.0.3) (2023-12-29)

**Note:** Version bump only for package @unleashit/configs

## [1.0.2](https://github.com/unleashit/npm-library/compare/@unleashit/configs@1.0.1...@unleashit/configs@1.0.2) (2023-12-29)

**Note:** Version bump only for package @unleashit/configs

## [1.0.1](https://github.com/unleashit/npm-library/compare/@unleashit/configs@1.0.0...@unleashit/configs@1.0.1) (2023-12-23)

**Note:** Version bump only for package @unleashit/configs

# [1.0.0](https://github.com/unleashit/npm-library/compare/@unleashit/configs@0.1.5...@unleashit/configs@1.0.0) (2023-08-01)

**Note:** Version bump only for package @unleashit/configs

## [0.1.5](https://github.com/unleashit/npm-library/compare/@unleashit/configs@0.1.4...@unleashit/configs@0.1.5) (2023-05-24)

**Note:** Version bump only for package @unleashit/configs

## [0.1.4](https://github.com/unleashit/npm-library/compare/@unleashit/configs@0.1.3...@unleashit/configs@0.1.4) (2023-05-24)

**Note:** Version bump only for package @unleashit/configs

## 0.1.3 (2023-05-09)

**Note:** Version bump only for package @unleashit/configs
