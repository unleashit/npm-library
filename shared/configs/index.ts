export * from './testConfig/jest.config';
export * from './testConfig/utils';
export * from './tsup/tsup.config';

// const { defineConfig } = require('tsup');
//
// const env = process.env.NODE_ENV;
//
// exports.tsup = defineConfig({
//   splitting: true,
//   clean: env === 'production', // clean up the dist folder
//   dts: true, // generate dts files
//   format: ['cjs', 'esm'], // generate cjs and esm files
//   minify: env === 'production',
//   bundle: true,
//   skipNodeModulesBundle: true,
//   entryPoints: ['src/index.ts'],
//   target: 'es2020',
//   sourcemap: env !== 'production',
//   outDir: 'dist',
//   entry: ['src/**/*.{ts,tsx}'], // include all files under src
// });

// const { tsup } = require('./tsup/tsup.config.js');
//
// exports.tsup = tsup;
