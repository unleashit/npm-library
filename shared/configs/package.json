{
  "name": "@unleashit/configs",
  "version": "1.0.6",
  "main": "index.ts",
  "module": "index.ts",
  "types": "./index.ts",
  "description": "Development dependencies for npm-library",
  "private": true,
  "scripts": {},
  "license": "MIT",
  "keywords": [
    "login",
    "react login",
    "responsive",
    "custom fields",
    "react",
    "react components",
    "npm packages",
    "npm components",
    "ui components",
    "ui library",
    "monorepo",
    "lerna",
    "typescript"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unleashit/npm-library.git"
  },
  "homepage": "https://github.com/unleashit/npm-library/tree/master/packages/dev",
  "bugs": {
    "url": "https://github.com/unleashit/npm-library/issues"
  },
  "devDependencies": {
    "react": "^18.2.0",
    "tsup": "^8.0.1"
  }
}
