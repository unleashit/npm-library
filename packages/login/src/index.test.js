import { mount } from 'enzyme';
import * as React from 'react';

// eslint-disable-next-line import/no-relative-packages
import { changeVal, nextTick } from '@unleashit/configs';
import LoginContainer from '.';

describe('<LoginContainer />', () => {
  let wrapper;
  const props = {
    loginHandler: () => jest.fn({ errors: {} }),
    onSuccess: () => jest.fn(),
  };

  beforeEach(() => {
    wrapper = mount(<LoginContainer {...props} />);
  });

  it('renders without crashing', () => {
    expect(wrapper.find('.unl-login__container')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  describe('validation', () => {
    it('fields can be updated and validated', () => {
      wrapper
        .find('input[name="email"]')
        .simulate('change', changeVal('email', 'test@test.com'));
      wrapper
        .find('input[name="password"]')
        .simulate('change', changeVal('password', '12345678'));

      expect(wrapper.find('input[name="email"]').props().value).toEqual('test@test.com');
      const errors = wrapper.find('.unl-login__error-message');
      expect(errors).toHaveLength(0);
    });

    it('displays correct email validation errors', async () => {
      const emailInput = wrapper.find('input[name="email"]');
      const passwordInput = wrapper.find('input[name="password"]');
      const form = wrapper.find('form');

      // email can't be empty
      passwordInput.simulate('change', changeVal('password', 'goodpassword'));
      form.simulate('submit', { preventDefault: () => undefined });
      await nextTick();
      wrapper.update();
      let emailError = wrapper.find('.unl-login__error-message').at(0);
      expect(emailError.text()).toEqual('email is a required field');

      // email must be a valid email
      emailInput.simulate('change', changeVal('email', 'bademail'));
      passwordInput.simulate('change', changeVal('password', 'goodpassword'));
      form.simulate('submit', { preventDefault: () => undefined });
      await nextTick();
      wrapper.update();
      emailError = wrapper.find('.unl-login__error-message').at(0);
      expect(emailError.text()).toEqual('email must be a valid email');
    });

    it('displays correct password validation errors', async () => {
      const emailInput = wrapper.find('input[name="email"]');
      const passwordInput = wrapper.find('input[name="password"]');
      const form = wrapper.find('form');

      // password field can't be empty
      emailInput.simulate('change', changeVal('email', 'good@email.com'));
      form.simulate('submit', { preventDefault: () => undefined });
      await nextTick();
      wrapper.update();
      let passwordError = wrapper.find('.unl-login__error-message').at(0);
      expect(passwordError.text()).toEqual('password is a required field');

      // password must be at least 8 chars
      emailInput.simulate('change', changeVal('email', 'good@email.com'));
      passwordInput.simulate('change', changeVal('password', '123'));
      form.simulate('submit', { preventDefault: () => undefined });
      await nextTick();
      wrapper.update();
      passwordError = wrapper.find('.unl-login__error-message').at(0);
      expect(passwordError.text()).toEqual('password must be at least 8 characters');
    });
  });
});
