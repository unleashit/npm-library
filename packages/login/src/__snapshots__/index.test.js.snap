// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> renders without crashing 1`] = `
<WithFormik(Function)
  loginHandler={[Function]}
  onSuccess={[Function]}
>
  <Formik
    enableReinitialize={false}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "email": "",
        "password": "",
        "serverAuth": "",
      }
    }
    isInitialValid={false}
    loginHandler={[Function]}
    onSubmit={[Function]}
    onSuccess={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <Component
      dirty={false}
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "email": "",
          "password": "",
          "serverAuth": "",
        }
      }
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      loginHandler={[Function]}
      onSuccess={[Function]}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "email": "",
          "password": "",
          "serverAuth": "",
        }
      }
    >
      <div
        className="unl-login__container"
      >
        <Component
          signupUrl="/signup"
          theme={Object {}}
        >
          <div
            className="unl-login__header"
          >
            <h2>
              Login
            </h2>
            <p>
              Don
              '
              t have an account yet? 
              <a
                href="/signup"
              >
                Sign up
              </a>
            </p>
          </div>
        </Component>
        <Form
          className="unl-login__form"
        >
          <Component
            className="unl-login__form"
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialValues": Object {
                  "email": "",
                  "password": "",
                  "serverAuth": "",
                },
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setError": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "status": undefined,
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validate": undefined,
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validationSchema": [Function],
                "values": Object {
                  "email": "",
                  "password": "",
                  "serverAuth": "",
                },
              }
            }
          >
            <form
              className="unl-login__form"
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <FormikConnect(FieldInner)
                component={[Function]}
                componentName="login"
                cssModuleStyles={Object {}}
                name="email"
                type="text"
              >
                <FieldInner
                  component={[Function]}
                  componentName="login"
                  cssModuleStyles={Object {}}
                  formik={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialValues": Object {
                        "email": "",
                        "password": "",
                        "serverAuth": "",
                      },
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setError": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validate": undefined,
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validationSchema": [Function],
                      "values": Object {
                        "email": "",
                        "password": "",
                        "serverAuth": "",
                      },
                    }
                  }
                  name="email"
                  type="text"
                >
                  <CustomInput
                    componentName="login"
                    cssModuleStyles={Object {}}
                    field={
                      Object {
                        "name": "email",
                        "onBlur": [Function],
                        "onChange": [Function],
                        "value": "",
                      }
                    }
                    form={
                      Object {
                        "dirty": false,
                        "errors": Object {},
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleReset": [Function],
                        "handleSubmit": [Function],
                        "initialValues": Object {
                          "email": "",
                          "password": "",
                          "serverAuth": "",
                        },
                        "isSubmitting": false,
                        "isValid": false,
                        "isValidating": false,
                        "registerField": [Function],
                        "resetForm": [Function],
                        "setError": [Function],
                        "setErrors": [Function],
                        "setFieldError": [Function],
                        "setFieldTouched": [Function],
                        "setFieldValue": [Function],
                        "setFormikState": [Function],
                        "setStatus": [Function],
                        "setSubmitting": [Function],
                        "setTouched": [Function],
                        "setValues": [Function],
                        "status": undefined,
                        "submitCount": 0,
                        "submitForm": [Function],
                        "touched": Object {},
                        "unregisterField": [Function],
                        "validateField": [Function],
                        "validateForm": [Function],
                        "validateOnBlur": true,
                        "validateOnChange": true,
                        "values": Object {
                          "email": "",
                          "password": "",
                          "serverAuth": "",
                        },
                      }
                    }
                    type="text"
                  >
                    <div
                      className="unl-login__form-group"
                    >
                      <label
                        className="unl-login__label"
                        htmlFor="login-form-email"
                      >
                        Email
                      </label>
                      <input
                        className="unl-login__input "
                        id="login-form-email"
                        name="email"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                    </div>
                  </CustomInput>
                </FieldInner>
              </FormikConnect(FieldInner)>
              <FormikConnect(FieldInner)
                component={[Function]}
                componentName="login"
                cssModuleStyles={Object {}}
                name="password"
                type="password"
              >
                <FieldInner
                  component={[Function]}
                  componentName="login"
                  cssModuleStyles={Object {}}
                  formik={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialValues": Object {
                        "email": "",
                        "password": "",
                        "serverAuth": "",
                      },
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setError": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validate": undefined,
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validationSchema": [Function],
                      "values": Object {
                        "email": "",
                        "password": "",
                        "serverAuth": "",
                      },
                    }
                  }
                  name="password"
                  type="password"
                >
                  <CustomInput
                    componentName="login"
                    cssModuleStyles={Object {}}
                    field={
                      Object {
                        "name": "password",
                        "onBlur": [Function],
                        "onChange": [Function],
                        "value": "",
                      }
                    }
                    form={
                      Object {
                        "dirty": false,
                        "errors": Object {},
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleReset": [Function],
                        "handleSubmit": [Function],
                        "initialValues": Object {
                          "email": "",
                          "password": "",
                          "serverAuth": "",
                        },
                        "isSubmitting": false,
                        "isValid": false,
                        "isValidating": false,
                        "registerField": [Function],
                        "resetForm": [Function],
                        "setError": [Function],
                        "setErrors": [Function],
                        "setFieldError": [Function],
                        "setFieldTouched": [Function],
                        "setFieldValue": [Function],
                        "setFormikState": [Function],
                        "setStatus": [Function],
                        "setSubmitting": [Function],
                        "setTouched": [Function],
                        "setValues": [Function],
                        "status": undefined,
                        "submitCount": 0,
                        "submitForm": [Function],
                        "touched": Object {},
                        "unregisterField": [Function],
                        "validateField": [Function],
                        "validateForm": [Function],
                        "validateOnBlur": true,
                        "validateOnChange": true,
                        "values": Object {
                          "email": "",
                          "password": "",
                          "serverAuth": "",
                        },
                      }
                    }
                    type="password"
                  >
                    <div
                      className="unl-login__form-group"
                    >
                      <label
                        className="unl-login__label"
                        htmlFor="login-form-password"
                      >
                        Password
                      </label>
                      <input
                        className="unl-login__input "
                        id="login-form-password"
                        name="password"
                        onBlur={[Function]}
                        onChange={[Function]}
                        type="password"
                        value=""
                      />
                    </div>
                  </CustomInput>
                </FieldInner>
              </FormikConnect(FieldInner)>
              <button
                className="unl-login__button"
                type="submit"
              >
                Login
              </button>
              <div
                className="unl-login__forgot-password-link"
              >
                <a
                  href="/forgot-password"
                >
                  Forgot your password?
                </a>
              </div>
            </form>
          </Component>
        </Form>
      </div>
    </Component>
  </Formik>
</WithFormik(Function)>
`;
